NDTensors/ext/NDTensorsAMDGPUExt/append.jl

```diff
@@ -1,5 +1,4 @@
-using GPUArraysCore: @allowscalar
-using AMDGPU: ROCArray
+using AMDGPU: @allowscalar, ROCArray
```

Comment:
I would prefer loading `@allowscalar` from `GPUArraysCore` if possible to make it clearer where it is defined, was there an issue with that?

Reply:
Hi Matt, I was writing you a message out about this. I am running into an issue that perhaps you know how to solve. My understanding is that when we use `GPUArraysCore: @allowscalar` in our extensions, Julia requires that we have `GPUArraysCore` as a `[deps]` of `NDTensors`. 
If I have `GPUArraysCore` in `[deps]`, `[weakdeps]`, `[extensions]` and `[extras]` header and have `using GPUArraysCore: @allwoscalar` in the extensions, Julia [ignores the package](https://pkgdocs.julialang.org/dev/creating-packages/#Transition-from-normal-dependency-to-extension) as a `[dep]` and I runs into an issue compiling the extension
```
Precompiling NDTensorsMetalExt
        Info Given NDTensorsMetalExt was explicitly requested, output will be shown live 
ERROR: LoadError: ArgumentError: Package NDTensorsMetalExt does not have GPUArraysCore in its dependencies:
```
If I remove `GPUArraysCore` from `[weakdeps]` and `[extras]` but keep it in the `[extensions]` category,  the package manager fails to resolve the manifest
```
(jl_RMaKFD) pkg> resolve
  No Changes to `/private/tmp/jl_RMaKFD/Project.toml`
ERROR: KeyError: key "GPUArraysCore" not found
Stacktrace:
```
and if I remove it from `[extensions]` then the `NDTensorsGPUArraysExt` file does not load. 
I could keep it as a `[deps]` only, keep the extensions folder and just include the module loke we do with the `src/lib` modules?
The proposed solution in this PR is the opposite of this where I remove `GPUArraysCore` as a dependency and make it only an extension.

Reply:
If you are only loading it inside of a package extension my understanding was that it does not need to be listed under `[deps]`.

Reply:
That's definitely a solution but it just feels like circumventing an issue that shouldn't be an issue and we should try to figure out a proper solution since I'm sure this will come up again in the future. I'm certain I've defined package extensions with multiple weak dependencies but I can't recall which ones right now.

I believe you can list multiple weak dependencies of a package extension though I think I tried that and may have run into trouble with it. Maybe this post helps: https://discourse.julialang.org/t/using-multiple-packages-in-extensions-for-julia-1-9/98237?

It seems like the conclusion there was to list `GPUArraysCore` under `[deps]`, which is ok since it will be installed but not loaded unless the package extension using it is loaded. That is a bit unsatisfying but may be the best we can do if it doesn't work to list both `GPUArraysCore` and `AMDGPU`, `CUDA`, etc. as weakdeps of the corresponding package extension.

Reply:
Note that in the example code in this section of the docs: https://pkgdocs.julialang.org/v1/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions) it says:
```
[extensions]
# name of extension to the left
# extension dependencies required to load the extension to the right
# use a list for multiple extension dependencies
PlottingContourExt = "Contour"
```
so you can list multiple extension dependencies but I'm not sure what happens if you do that. Have you tried that?

Reply:
Ahh thank you, this was something I had meant to try but I had forgotten! I added `GPUArraysCore` to the `weakdeps` list for the GPU extensions and that fixed the problem
```
NDTensorsMetalExt = ["GPUArraysCore", "Metal"]
```
Now I can include `GPUArraysCore` in the extensions and I do not have it listed in the `[deps]` section of `NDTensors.